### PR TITLE
refactor(artwork): decouple and defer RelatedWorks query

### PR DESCRIPTION
**Pull request by anandaroop**
> **Warning**
> All good now. ~Depends on https://github.com/artsy/metaphysics/pull/4545, thus the current Relay error in CI. Once that MP schema update makes it to Force, this PR will need a rebase. Otherwise this is ready for review.~ 

The type of this PR is: **Refactor**

This PR solves [FX-4454]

### Description

This follows some investigations into Elasticsearch cluster issues (see [ticket](https://artsyproduct.atlassian.net/browse/FX-4454) for links to Slack threads) 

Earlier  we tried to improve things by making the big `OtherWorks` query lazier (https://github.com/artsy/force/pull/11304)

This follows up by moving the expensive `RelatedWorks` rail out of the `OtherWorks` component altogether — so that it can be deferred further, and so that its latency and errors won't affect the several other `OtherWorks` rails.

From a UX perspective it is almost a pure refactor, though it is possible we are discarding some seldom-if-at-all-seen tabbed UIs that _might_ have appeared under the **Related works** heading. Now there is just one set of artworks there, corresponding to the `main` related layer in Gravity, ultimately [this](https://github.com/artsy/gravity/blob/main/app/models/search/layers/main_layer.rb).

before after

cc @artsy/velocity-engineers 

[FX-4454]: https://artsyproduct.atlassian.net/browse/FX-4454?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Apps/Artwork/Components/OtherWorks/index.tsx [hunk 1]
```diff
@@ -123,22 +118,14 @@ export const OtherWorks = track()(
...
-
-            <RelatedWorksArtworkGridRefetchContainer artwork={artwork} />
-          </>
-        )}
-
+        {/*
+        TODO: Appears to never be true? Maybe this was supposed to match
+        context.__typename === "Sale"
+        or contextGrids[i] === "ArtworkContextAuction"
+        */}
```
Comment (anandaroop) [inline src/Apps/Artwork/Components/OtherWorks/index.tsx:125]: Little breadcrumb for the future — this was unrelated to my refactor, but in studying this code I began to think that the `OtherAuctionsQueryRenderer` below never actually gets rendered? 🤷🏽

File: src/Apps/Artwork/Components/OtherWorks/index.tsx [hunk 2]
```diff
@@ -152,7 +139,7 @@ export const OtherWorksFragmentContainer = createFragmentContainer(
   {
     artwork: graphql`
       fragment OtherWorks_artwork on Artwork {
-        contextGrids {
+        contextGrids(includeRelatedArtworks: false) {
```
Comment (anandaroop) [inline src/Apps/Artwork/Components/OtherWorks/index.tsx:142]: Extracting the `RelatedWorks` rail from the frontend in Force into its own QueryRenderer wasn't sufficient to move the expensive related artworks query out of `OtherWorks`. 

We needed to additionally deal with the opinionated backend logic in MP that _also_ fetched those related works in order to plop them into this `contextGrids` response.

Now we can opt out of that expensive step, thanks to https://github.com/artsy/metaphysics/pull/4545

File: src/Apps/Artwork/Components/RelatedWorks/index.tsx [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+        contextModule={ContextModule.relatedWorksRail}
+        artworks={artworksConnection}
+        columnCount={[2, 3, 4, 4]}
+        onBrickClick={() =>
+          trackEvent({
+            type: DeprecatedSchema.Type.ArtworkBrick,
+            action_type: DeprecatedSchema.ActionType.Click,
+            context_module: DeprecatedSchema.ContextModule.RelatedWorks,
+          })
+        }
```
Comment (anandaroop) [inline src/Apps/Artwork/Components/RelatedWorks/index.tsx:48]: Confirmed that this event flows through to the Segment debugger with the same shape as the old one:

```
analytics.track('Click', {
  context_module: 'RelatedWorks',
  type: 'Artwork brick'
});
```

**Comment by joeyAghion** [comment]
🙏

**Comment by anandaroop** [comment]
Updated now that the related MP schema change is in Force. I'll take the WIP off now, feel free to give this a look if you haven't.

File: src/Apps/Artwork/Components/RelatedWorks/index.tsx [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+      fragment RelatedWorks_artwork on Artwork {
+        slug
+        title
+        layer(id: "main") {
+          name
+          artworksConnection(first: 8) {
+            ...ArtworkGrid_artworks
+            # Used to check for content
+            edges {
+              node {
```
Comment (dblandin) [inline src/Apps/Artwork/Components/RelatedWorks/index.tsx:79]: **suggestion / non-blocking**: If we're arbitrarily fetching `node.slug` so we can check that we have edges to render, we could use `__typename` instead.

```
edges {
  __typename
}
```

**Review by dblandin — APPROVED** [review]
Nice! LGTM 💯

File: src/Apps/Artwork/Components/RelatedWorks/index.tsx (hunk 4 above)
Comment (mzikherman) [reply src/Apps/Artwork/Components/RelatedWorks/index.tsx:79]: Another alternative, most/all connections offer a `totalCount` connection-level field that might be useful too!

Comment (anandaroop) [reply src/Apps/Artwork/Components/RelatedWorks/index.tsx:79]: Good thoughts! I originally cargo-culted this snippet [from the old implementation](https://github.com/artsy/force/blob/931716dc511598174592480b178f93301cdcbbbb/src/Apps/Artwork/Components/OtherWorks/RelatedWorksArtworkGrid.tsx#L144-L149), but found that `artwork.layer.artworksConnection` didn't support `totalCount` and that requesting only `__typename` didn't change the upstream fetching behavior. So I'm gonna ahead and give this a merge.

**Pull request merged**